modules/sdk/src/main/scala/org/tessellation/sdk/domain/Daemon.scala

```diff
@@ -14,4 +18,16 @@ object Daemon {
     def start: F[Unit] = S.supervise(thunk).void
   }
 
+  def periodic[F[_]: Async](thunk: F[Unit], sleepTime: FiniteDuration)(implicit S: Supervisor[F]): Daemon[F] =
+    new Daemon[F] {
+      def start: F[Unit] =
+        S.supervise(
+          Stream
+            .awakeEvery(sleepTime)
+            .evalMap(_ => thunk)
+            .compile
+            .drain
+        ).void
+    }
+
```

Comment:
what about error handling? a single error in thunk will cancel whole periodic

Reply:
@marcinwadon @cngo-github 
Good point. I have a test app that confirms that is the case.

However, if it's a concern here isn't it also a concern in the rest of the codebase? This is just a generalization of what's already duplicated in a few places.

As a solution, what about changing line 27 to
```
 .evalMap(_ => thunk.handleError(e => println(e.getMessage)))
```
where we replace `println` with `logger.error`?
I confirmed that in my app it continues with the stream on error.

Reply:
Yep, this will log the error and continue the stream. and indeed, it's a concern in the rest of the codebase so we should take care of it. Anyway, I've created a global error handling:
![image](https://github.com/Constellation-Labs/tessellation/assets/4959931/a71b4ffa-589e-436e-aa56-708c48b2409e)
so in case we forgot, at least we can see the error before the service dies

Reply:
Added error handling to the daemon.